File: week2-arrays/problem-set-2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <math.h>

string count_everything(string text, int *letters, int *words, int *sentences);

int main(void)
{
    string text = get_string("Text: ");
    int letters = 0;
    int words = 0;
    int sentences = 0;
    count_everything(text, &letters, &words, &sentences);
    float L = (float) letters / words * 100.0;
    float S = (float) sentences / words * 100.0;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int grade = round(index);
    if (grade < 1)
    {
        printf("Before Grade 1\n");
        return 0;
    }
    if (grade >= 16)
    {
        printf("Grade 16+\n");
        return 0;
    }
    printf("Grade %i\n", grade);
}

string count_everything(string text, int *letters, int *words, int *sentences)
{
    string res = text;
    while (*text)
    {
        if (isalpha(*text))
        {
            (*letters)++;
        }
        if (*text == ' ')
        {
            (*words)++;
        }
        if (*text == '?' || *text == '!' || *text == '.')
        {
            (*sentences)++;
        }
        text++;
    }
    (*words)++;
    return res;
}

File: week6-python/problem-set/sentimental-readability/readability.py
def main():
    text = input("Text: ")
    # count words, sentences and letters
    words = text.split(" ")
    words_count = len(words)
    sentence_delimiters = [".", "?", "!"]
    sentences_count = 0
    for char in text:
        if char in sentence_delimiters:
            sentences_count += 1
    letters_count = 0
    for char in text:
        if char.isalpha():
            letters_count += 1
    L = (letters_count / words_count) * 100
    S = (sentences_count / words_count) * 100
    index = 0.0588 * L - 0.296 * S - 15.8
    index = round(index)
    if index < 1:
        print(f"Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")

main()